monai/engines/evaluator.py

```diff
@@ -288,7 +299,9 @@ def _iteration(self, engine: SupervisedEvaluator, batchdata: dict[str, torch.Ten
             kwargs: dict = {}
         else:
             inputs, targets, args, kwargs = batch
-
+        if self.compile:
+            inputs = torch.Tensor(inputs) if isinstance(inputs, MetaTensor) else inputs
+            targets = torch.Tensor(targets) if isinstance(targets, MetaTensor) else targets
```

Comment:
Is there a specific reason MetaTensor isn't compatible with compilation? I haven't tried it myself and a bit out of the loop on this. If we cast to regular tensor like this the outputs from networks lose their relationship with the original inputs, eg. filenames and such in the metadata.

Reply:
Will create a simple repro and raise a ticket in the torch repo, send you the link later.

Reply:
https://github.com/pytorch/pytorch/issues/117026